File: Python/1049.py
# -*- coding: utf-8 -*-
f1 = input()
f2 = input()
f3 = input()
        
if ("vertebrado" == f1):
    if ("ave" == f2):
        if ("carnivoro" == f3):
            print("aguia")
                
        elif ("onivoro" == f3):
            print("pomba")
                
            
    elif ("mamifero" == f2):
        if ("onivoro" == f3):
            print("homem")
                
        elif ("herbivoro" == f3):
            print("vaca")
           

elif ("invertebrado" == f1):
    if ("inseto" == f2):
        if ("hematofago" == f3):
            print("pulga")
                
        elif ("herbivoro" == f3):
            print("lagarta")
                
           
    elif ("anelideo" == f2):
        if ("hematofago" == f3):
            print("sanguessuga")
                
        elif ("onivoro" == f3):
            print("minhoca")

File: Java/1049.java
import java.io.IOException;
import java.util.Scanner;

public class Main {
 
    public static void main(String[] args) throws IOException {
        Scanner leitor = new Scanner(System.in);
        
        String f1 = leitor.next();
        String f2 = leitor.next();
        String f3 = leitor.next();
        
        if ("vertebrado".equals(f1)) {
            if ("ave".equals(f2)) {
                if ("carnivoro".equals(f3)) {
                    System.out.println("aguia");
                }
                else if ("onivoro".equals(f3)) {
                    System.out.println("pomba");
                }
            }
            else if ("mamifero".equals(f2)) {
                if ("onivoro".equals(f3)) {
                    System.out.println("homem");
                }
                else if ("herbivoro".equals(f3)) {
                    System.out.println("vaca");
                }
            }
        }
        else if ("invertebrado".equals(f1)) {
            if ("inseto".equals(f2)) {
                if ("hematofago".equals(f3)) {
                    System.out.println("pulga");
                }
                else if ("herbivoro".equals(f3)) {
                    System.out.println("lagarta");
                }
            }
            else if ("anelideo".equals(f2)) {
                if ("hematofago".equals(f3)) {
                    System.out.println("sanguessuga");
                }
                else if ("onivoro".equals(f3)) {
                    System.out.println("minhoca");
                }
            }
        }
    }
 
}